pages/login/index.js

```diff
@@ -1,37 +1,163 @@
-import "/common/index.js";
+import {} from "/common/index.js";
 
-const icons = Object.freeze({
-	["text"]: "/assets/icons/visible.svg",
-	["password"]: "/assets/icons/invisible.svg",
+const inputs = new Set(
+	document.querySelectorAll("input")
+);
+
+const selector = new Map([
+	["form", document.querySelector("form")],
+	["button", document.querySelector("form button")],
+]);
+
+function validate(input)
+{
+	const message = input["message"]();
+	// display error
+	input.closest(".wrapper").dataset["error"] = message ?? "null";
+	// toggle button
+	selector.get("button").disabled = !(!message && ![...inputs.values()].find((element, index, array) => element === input ? message : element["message"]()));
```

Comment:
사용하지 않는 인자는 작성하지 않는 편이 일반적입니다
```suggestion
	selector.get("button").disabled = !(!message && ![...inputs.values()].find((element, index) => element === input ? message : element["message"]()));
```

Reply:
명시를 중요하게 생각해서 사용해 봤습니다만, JavaScript 라는 언어의 특징을 앞으로는 보다 수용하려 노력해보도록 하겠습니다.